wdl-analysis/src/handlers/document_symbol.rs

```diff
@@ -0,0 +1,380 @@
+    // to complete and call `structs`, `tasks` and `workflow` on
+    // `analysis::Document`. Doing so will break the outline while the user is
+    // typing.
+    for item in ast.items() {
+        match item {
+            DocumentItem::Workflow(workflow) => {
+                symbols.push(workflow_to_symbol(uri, &workflow, &lines)?);
+            }
+            DocumentItem::Task(task) => {
+                symbols.push(task_to_symbol(uri, &task, &lines)?);
+            }
+            DocumentItem::Struct(s) => {
+                symbols.push(struct_to_symbol(uri, &s, &lines)?);
+            }
+            DocumentItem::Import(ns) => {
+                symbols.push(import_to_symbol(uri, &ns, &lines)?);
+            }
+        }
+    }
+
+    Ok(Some(DocumentSymbolResponse::Nested(symbols)))
+}
+
+/// Converts a [`ImportStatement`] to a [`DocumentSymbol`]
+fn import_to_symbol(
+    uri: &Url,
+    import: &ImportStatement,
+    lines: &std::sync::Arc<line_index::LineIndex>,
+) -> Result<DocumentSymbol> {
+    let (name, selection_span) = import.namespace().unwrap_or_else(|| {
+        (
+            import.uri().text().unwrap().text().to_string(),
+            import.uri().span(),
+        )
+    });
+
+    Ok(DocumentSymbol {
+        name,
+        detail: Some(import.uri().text().unwrap().text().to_string()),
+        kind: SymbolKind::NAMESPACE,
+        range: common::location_from_span(uri, import.span(), lines)?.range,
+        selection_range: common::location_from_span(uri, selection_span, lines)?.range,
+        children: None,
+        tags: None,
+        #[allow(deprecated)]
+        deprecated: None,
+    })
+}
+
+/// Converts a [`WorkflowDefinition`] to a [`DocumentSymbol`]
+fn workflow_to_symbol(
+    uri: &Url,
+    workflow: &WorkflowDefinition,
+    lines: &std::sync::Arc<line_index::LineIndex>,
+) -> Result<DocumentSymbol> {
+    let mut children = Vec::new();
+
+    for item in workflow.items() {
+        match item {
+            WorkflowItem::Input(section) => {
+                children.extend(input_section_to_symbols(uri, &section, lines)?)
+            }
+            WorkflowItem::Output(section) => {
+                children.extend(output_section_to_symbols(uri, &section, lines)?)
+            }
+            WorkflowItem::Declaration(decl) => {
+                children.push(bound_decl_to_symbol(uri, &decl, lines)?)
+            }
+            WorkflowItem::Call(call) => children.push(call_to_symbol(uri, &call, lines)?),
+            WorkflowItem::Conditional(cond) => {
+                children.push(conditional_to_symbol(uri, &cond, lines)?)
+            }
+            WorkflowItem::Scatter(scatter) => {
+                children.push(scatter_to_symbol(uri, &scatter, lines)?)
+            }
+            _ => {}
+        }
+    }
+
+    Ok(DocumentSymbol {
+        name: workflow.name().text().to_string(),
+        detail: Some("workflow".to_string()),
+        kind: SymbolKind::MODULE,
```

Comment:
My gut is that this should be `SymbolKind::FUNCTION`.

Reply:
Honestly, I think the latter looks much better, even with the "function in function" feel. The alternative was to have workflows look like imported modules alongside the actual imports, which seemed even stranger.